File: Data Structures & Algorithms/gas-station/submission-1.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        # ans=-1
        n=len(gas)
        tank=0
        si=0
        if (sum(gas)<sum(cost)):
            return -1
        for i in range (n):
           tank+=gas[i]-cost[i]
           if tank<0:
            si=i+1
            tank=0
        return si

File: Data Structures & Algorithms/non-cyclical-number/submission-1.cpp
class Solution {
public:
    bool isHappy(int n) {
     set<int>s;
     while(s.find(n)==s.end())   {
     s.insert(n);
     int sum=0;
     while(n>0){
        int r=n%10;
        n/=10;
        sum+=r*r;
        cout<<sum<<endl;

     }
     if(sum==1)return true;
     n=sum;
     }
    return false;
    }
};
